aws-xray-recorder-sdk-core/src/main/java/com/amazonaws/xray/entities/TraceID.java

```diff
@@ -22,37 +22,7 @@
 
 public class TraceID {
 
-    private static final int TRACE_ID_LENGTH = 35;
-    private static final int TRACE_ID_DELIMITER_INDEX_1 = 1;
-    private static final int TRACE_ID_DELIMITER_INDEX_2 = 10;
-
-    private static final char VERSION = '1';
-    private static final char DELIMITER = '-';
-
-    private BigInteger number;
-    private long startTime;
-
-    /**
-     * @deprecated Use {@link #create()}.
-     */
-    @Deprecated
-    public TraceID() {
-        this(Instant.now().getEpochSecond());
-    }
-
-    /**
-     * @deprecated Use {@link #create()}.
-     */
-    @Deprecated
-    public TraceID(long startTime) {
-        number = new BigInteger(96, ThreadLocalStorage.getRandom());
-        this.startTime = startTime;
-    }
-
-    private TraceID(long startTime, BigInteger number) {
-        this.startTime = startTime;
-        this.number = number;
-    }
+    private static final TraceID INVALID = new TraceID(0, BigInteger.ZERO);
```

Comment:
Yeah thought about this a bit and my leaning was that it might be a good thing for this to be invalid downstream too as this is only used when a segment isn't emitted - if it got emitted for some wrong reason, filtering it out because of bad timestamp might actually be better.

I looked at node and go SDKs and didn't see any issues with this, and wouldn't expect an SDK to do any timestamp validation as that's a backend requirement. So still lean towards 0 but let me know if there's anything else to be worried about.

Reply:
Will discuss offline

Reply:
Resolved offline; Ok as is